### ISSUE TITLE
change number of channels in table DSP objects

### ISSUE THREAD

**Issue by Spacechild1** [pure-data/pure-data#2025]
Currently, the number of channels in all the table DSP objects fixed - determined by the creation arguments - and the "set" message can only override the provided tables. There is no way to actually change the number of channels!

`tabsend~` and `tabreceive~` are conceptually similar to `send~` and `receive~` - which have the "channels" method. `tabwrite~` and `tabplay~` are basically in-memory versions of `readsf~` and `writesf~` - which will likely get multi-channel support in the future.

I think all the table DSP objects should support dynamic channel counts, for all the reasons mentioned in https://github.com/pure-data/pure-data/pull/1979.

Here is a simple suggestion: *allow "set" to change the number of channels*. For example, "set foo" -> 1 channel, "set foo bar" -> 2 channels, "set foo bar baz" -> 3 channels, etc. If @millerpuckette agrees in principle, I can make a PR.

Or does anyone have a better idea?

NB: if we change the behavior of the "set" message, we would need to do this *before* releasing Pd 0.54!

---

Side note: IIRC, Miller thinks about adding support for data structure arrays in all DSP table objects. This would enable the usage of multi-dimensional arrays for multi-channel operations. However, in that case there would be no point in restricting the object to a fixed number of channels, and neither should we do for "regular" graphical arrays.

**Comment by millerpuckette** [comment]
I think this can be left for later in a separate "channels" message.  Otherwise people migth cause undesired DSP re-sorts by accident.  (The documentation should clearly separate messages that cause re-sorting from others - but that should probably be part of a larger campaign to clearly mark messages that could disrupt audio flow (e.g., reading files).

**Comment by Spacechild1** [comment]
>  think this can be left for later in a separate "channels" message. 

Yes, that's also possible. In that case there is indeed no hurry.

**Comment by Spacechild1** [comment]
Closed via https://github.com/pure-data/pure-data/pull/2595
